src/test/java/service/impl/DataConverterImplTest.java

```diff
@@ -0,0 +1,110 @@
+package service.impl;
+
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import model.FruitTransaction;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import service.DataConverter;
+
+class DataConverterImplTest {
+    private static final String EMPTY_LINES_FILE = "src/main/java/resources/fileWithEmptyLines.csv";
+    private static final String INVALID_OPERATION = "a";
+    private static final String PROPER_HEADER = "type,fruit,quantity";
+    private static final String FIRST_PROPER_LINE = "b,apple,32";
+    private static final String SECOND_PROPER_LINE = "b,banana,12";
+    private static final String THIRD_PROPER_LINE = "s,orange,56";
+    private static final int NEGATIVE_QUANTITY = -9;
+    private static DataConverter converter;
+    private FruitTransaction firstTransaction;
+    private FruitTransaction secondTransaction;
+    private FruitTransaction thirdTransaction;
+
+    @BeforeAll
+    static void beforeAll() {
+        converter = new DataConverterImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        firstTransaction = new FruitTransaction("b", "apple", 32);
+        secondTransaction = new FruitTransaction("b", "banana", 12);
+        thirdTransaction = new FruitTransaction("s", "orange", 56);
+    }
+
+    @Test
+    void readValidFile_Ok() {
+        List<String> properData = Arrays.asList(
+                PROPER_HEADER, FIRST_PROPER_LINE, SECOND_PROPER_LINE, THIRD_PROPER_LINE);
+        List<FruitTransaction> actualList = converter.convertToTransaction(properData);
+        List<FruitTransaction> expectedList = List.of(
+                firstTransaction, secondTransaction, thirdTransaction);
+        assertEquals(expectedList, actualList);
+    }
+
+    @Test
+    void tryReadFileWithEmptyLines() {
+        List<String> data = readFromFile(EMPTY_LINES_FILE);
```

Comment:
Don’t  complicate this test case 
```suggestion
        List<String> data = List.of();
```